core/station/impl/src/models/account.rs

```diff
@@ -249,6 +250,35 @@ impl Account {
     }
 }
 
+pub enum BalanceQueryState {
+    StaleRefreshing,
+    Stale,
+    Fresh,
+}
+
+impl fmt::Display for BalanceQueryState {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            BalanceQueryState::StaleRefreshing => write!(f, "stale_refreshing"),
+            BalanceQueryState::Stale => write!(f, "stale"),
+            BalanceQueryState::Fresh => write!(f, "fresh"),
+        }
+    }
+}
+
+impl From<&AccountBalance> for BalanceQueryState {
+    fn from(balance: &AccountBalance) -> Self {
+        let balance_age_ms = crate::core::ic_cdk::api::time()
+            .saturating_sub(balance.last_modification_timestamp)
+            / 1000;
```

Comment:
time get's back the time in nanoseconds, to get the miliseconds you need to divide by `1_000_000`